src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -62,6 +70,58 @@
     private static final String TAG_STYLE = "style";
 
     /**
+     * A list of ItemStacks with needed items and their quantity.
+     * This list is a diff between itemsNeeded in AbstractEntityAiBasic and
+     * the players inventory and their hut combined.
+     * So look here for what is currently still needed
+     * to fulfill the workers needs.
+     * <p>
+     * Will be cleared on restart, be aware!
+     */
+    @NotNull
+    private List<ItemStack> itemsCurrentlyNeeded = new ArrayList<>();
+
+    /**
+     * A list which contains the position of all containers which belong to the worker building.
+     */
+    private final List<BlockPos> containerList = new ArrayList<>();
+
+    /**
+     * This flag tells if we need a shovel, will be set on tool needs.
+     */
+    private boolean needsShovel = false;
```

Comment:
We aren't tracking anything like that now, with just a boolean. And then later we end up converting to the Strings, so I think a list of strings would be better